src/components/CreatedByInput.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from 'react'
+import { type Identifier, useGetIdentity } from 'react-admin'
+import { useFormContext } from 'react-hook-form'
+
+interface Props { source?: string }
+
+export default function CreatedByInput(props: Props): React.ReactElement {
+  const { source = 'createdBy' } = props
+  const { data } = useGetIdentity()
+  const formContext = useFormContext()
+
+  useEffect(() => {
+    if (formContext !== null) {
+      const { setValue, getValues } = formContext
+
```

Comment:
@salony75way - we should probably popup an error dialog if there isn't an identity, it's probably a but that would want to spot ourselves, or get users to tell us about.